### ISSUE TITLE
Labeled Box

### ISSUE THREAD

**Issue by kgarwoodsdzwa** [conservationtechlab/animl-py#11]
The function drawBoundingBoxes within labeled box.ipynb within src/animl is a Jupyter Notebooks file format, but it is meant to be a .py function so that it can be an accessible tool. It should pass flake8 and pylint and have docstrings.

**Comment by iingram** [comment]
- Specifically, we have decided for the moment to call the module just `plotboxes.py`
- that module would, for the time being, only contain that one function.
- then we have to think about where to put an example script that encapsulates what the Jupyter notebook currently does.  Maybe in a folder called `example_applications` at the highest level of the repo? thoughts on that?

**Comment by srinidhi98** [comment]
Sounds good, putting in a separate folder(example_applications/example_script) in src/animl might do. and later include in README.

**Comment by iingram** [comment]
I think only package files should go in `src/animal`
I also just noticed that an examples folder has been created at the highest level of the repo so that seems like a good place to put this example script.
And, on that note, the example Jupyter notebooks likely belong there, too.

**Comment by iingram** [comment]
I know I said I would capture some of our (Nidhi and my) Zoom discussion things into this GitHub discussion.  Not sure I remember all the things :-). But here's some:

1) as you pointed out, beyond creating the example script that uses the box plotting tool we are going to need some way to provide some data (images and ((df and/or csv) or script that generates df and/or csv) for the script to work on.
2) ideally, eventually, we'd serve some example images from somewhere not in this repo (in order to keep this repo clean of heavy, non-version-control-needing binaries) and have a data-download script that gets that data that would be run by a user that wants to run these example scripts.  
3) Right now, some image data *has* been committed into the repo (later we should actually undo this and rebase to clean things up but let's leave that for another time) so we could just work with that to get started.
4) as sort of alluded to in 1), we would also want an example script that runs the whole AniML pipeline on a given set of data (with an example config file that is set to work with the example data downloaded with the data-download script) so that there would be no need to provide an example dataframe or csv for the `plotbox_example.py` script as these would just be automatically generated.
5) NOTABLY: that all said, I think our plan at the end of the discussion was to focus this feature branch and pull request on just creating the plot_boxes.py module in the package files and putting the function from the Jupyter notebook in that module and then (in a separate commit or series of commits) linting that module (our baseline is passing `flake8` and `pylint`). 

We would then address 1), 2), and 4) in follow-on issues/PRs.

So, in summary, @srinidhi98 just go ahead and do 5) on your feature branch (you might want to make a fresh one branching from main so that your test commits don't end up in the final history) and once it's ready, submit a PR listing me and Katie as reviewers since we are filling in while Kyra is gone.  You might make a "fake" csv file with a single row (or maybe even two rows) that references one/two of the images that are currently in the repo (box doesn't have to be in the right place since it is just fake testing data) so that you have something to test the function on with the fringe benefit that you'll get a sense of what is included in each row of the AniML pandas dataframe and thus the CSV.

**Comment by iingram** [comment]
Some quick notes on our meeting just now:

- use argparse module to get CSV filename to use for box plotting as command line argument
- example scripts (including this one) in `examples` at highest level of the repo
- when you do pylint on plotboxes.py, it may give a bunch of E1101 errors, for now easy way to mute those is:
     `pylint --disable=E1101 plot_boxes.py`
- not crucial at this stage, but the lab tries to stick to NumPy style docstrings
- however, is super great that you did already add a docstring for the `plot_boxes` function and provided some clarity on the arguments!
- i realized why pylint was throwing that error on the function docstring: the `"""`'s weren't indented :-)
- also just noticed that the branch "12_Labeled_box" is branching from the other branch you had created.  It would be best to checkout main and create a fresh issue #11 feature branch from there.  But for the moment, perhaps just work in the branch you've been working on and tomorrow we can take care of branching differently.

**Comment by iingram** [comment]
This issue was resolved by PR #12
